### PR TITLE
[chore] Fix Download of Filters

### PR DESCRIPTION
**Pull request by theBeginner86**
**Notes for Reviewers**

This PR ensures that binary WASM file is downloaded in the same format as it was uploaded.

https://github.com/meshery/meshery/assets/73700530/ca749ccf-7e65-409c-8b84-6a4aa0726075

This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@Abhishek-kumar09 we have an ongoing issue with the eslint checks.

File: server/handlers/error.go [hunk 1]
```diff
@@ -309,6 +310,11 @@ func ErrApplicationFailure(err error, obj string) error {
 func ErrApplicationSourceContent(err error, obj string) error {
 	return errors.New(ErrApplicationContentCode, errors.Alert, []string{"failed to ", obj, "the application content"}, []string{err.Error()}, []string{}, []string{})
 }
+
+func ErrDownloadWASMFile(err error, obj string) error {
+	return errors.New(ErrDownlaodWASMFileCode, errors.Alert, []string{"failed to ", obj, "the WASM file"}, []string{err.Error()}, []string{"Ensure that DB is not corrupted", "Ensure Remote Provider is working properly", "Ensure Meshery Server is working properly and connected to remote provider"}, []string{"Try restarting Meshery server"})
```
Comment (leecalcote) [inline server/handlers/error.go:315]: Thank you

**Comment by leecalcote** [comment]
I wonder if the remote provider tracks download events.

**Review by leecalcote — APPROVED** [review]

**Comment by theBeginner86** [comment]
> I wonder if the remote provider tracks download events.

Not, yet. 

> @Abhishek-kumar09 we have an ongoing issue with the eslint checks.

Fixed here: https://github.com/meshery/meshery/pull/8057

**Comment by theBeginner86** [comment]
> @Abhishek-kumar09 we have an ongoing issue with the eslint checks.

Fixed here: https://github.com/meshery/meshery/pull/8057

**Pull request merged**